distributions/aws/infra_configs/scripts/aws/route53.py

```diff
@@ -0,0 +1,159 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Dict
+import boto3
+import logging
+import random
+import string
+
+from botocore.exceptions import ClientError, WaiterError
+from typing import Dict
+
+logger = logging.getLogger(__name__)
+
+
+class Route53HostedZone:
+    """
+    Encapsulates route53 functions.
+    """
+
+    def __init__(
+        self,
+        domain: str,
+        id: str = None,
+        route53_client: Any = None,
+        region: str = "us-east-1",
+    ):
+        """
+        :param route53_client: A Boto3 route53 client.
+        :param domain: domain corresponding to this reference
+        """
+        self.route53_client = route53_client or boto3.client(
+            "route53", region_name=region
+        )
+        self.domain = domain
+        self.id = id
+
+    def create_zone(self) -> dict:
```

Comment:
How is this function will be used internally vs externally ?

Reply:
Can you provide more detail on this to help me understand? What is internal and external